tests/tape/test_tape.py

```diff
@@ -197,7 +197,7 @@ def test_tensor_process_queuing(self):
             C = A @ B
             D = qml.expval(C)
 
-        assert len(tape.queue) == 4
+        assert len(tape.queue) == 1
```

Comment:
Do we really want this behaviour? When running
```python
with QuantumTape() as tape:
    A = qml.PauliX(wires=0)
    B = qml.PauliZ(wires=1)
    C = A @ B
    D = qml.expval(C)
```
I expect the queue to have 4 operations, not 1.

Reply:
And I still don't understand why did the behaviour change. I thought the change in this PR didn't change any queuing behaviour.

Reply:
It changes queuing. It just doesn't change behaviour.

You "expect" 4 operations, because that's how we did it in the past. But do we need to keep doing it that way in the future? While that's what we now expect, is that just because what we've seen, not what actually makes sense.

Reply:
Okay, I was confused because I thought that `process_queue` removed the elements from the queue.

Honestly now it makes much more sense. Before we would have:
```pycon
>>> with QuantumTape() as tape:
...    A = qml.PauliX(wires=0)
...    B = qml.PauliZ(wires=1)
...    C = A @ B
...    D = qml.expval(C)
>>> len(tape.queue)
4
>>> len(tape)
1
```
which is confusing.

Reply:
Exactly. Instead of waiting for `process_queue` to remove operations, we remove them immediately.